app/components/announcement_banner/expanded_announcement_banner.tsx

```diff
@@ -0,0 +1,136 @@
+import {Text, View} from 'react-native';
+import Button from 'react-native-button';
+import {ScrollView} from 'react-native-gesture-handler';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import {dismissAnnouncement} from '@actions/local/systems';
+import FormattedText from '@components/formatted_text';
+import Markdown from '@components/markdown';
+import {Screens} from '@constants';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {dismissBottomSheet} from '@screens/navigation';
+import {buttonBackgroundStyle, buttonTextStyle} from '@utils/buttonStyles';
+import {getMarkdownTextStyles, getMarkdownBlockStyles} from '@utils/markdown';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+type Props = {
+    allowDismissal: boolean;
+    bannerText: string;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+        },
+        scrollContainer: {
+            flex: 1,
+            marginTop: 12,
+            marginBottom: 24,
+        },
+        baseTextStyle: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'Regular'),
+        },
+        title: {
+            color: theme.centerChannelColor,
+            ...typography('Heading', 600, 'SemiBold'),
+        },
+    };
+});
+
+const close = () => {
+    dismissBottomSheet();
+};
+
+const ExpandedAnnouncementBanner = ({
+    allowDismissal,
+    bannerText,
+}: Props) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+    const isTablet = useIsTablet();
+    const intl = useIntl();
+    const insets = useSafeAreaInsets();
+
+    const dismissBanner = useCallback(() => {
+        dismissAnnouncement(serverUrl, bannerText);
+        close();
+    }, [bannerText]);
+
+    const buttonStyles = useMemo(() => {
+        return {
+            okay: {
+                button: buttonBackgroundStyle(theme, 'lg', 'primary'),
+                text: buttonTextStyle(theme, 'lg', 'primary'),
+            },
+            dismiss: {
+                button: [{marginTop: 10}, buttonBackgroundStyle(theme, 'lg', 'link')],
+                text: buttonTextStyle(theme, 'lg', 'link'),
+            },
+        };
+    }, [theme]);
+
+    const containerStyle = useMemo(() => {
+        return [style.container, {marginBottom: insets.bottom + 10}];
+    }, [style, insets.bottom]);
+
+    return (
+        <View style={containerStyle}>
+            {!isTablet && (
+                <Text style={style.title}>
+                    {intl.formatMessage({
+                        id: 'mobile.announcement_banner.title',
+                        defaultMessage: 'Announcement',
+                    })}
+                </Text>
+            )}
+            <ScrollView
```

Comment:
if the content is large enough, should we actually allow the bottom sheet to expand to `90%` of the screen or are we ok with only one snap point?

Reply:
The problem is that, as far as I know, we cannot know how big the text is going to be. Therefore, with "small text" we have lots of white space, with "long texts" we have a scroll view. We can try to infer from the number of characters, but that can go wrong really fast. If we want something more dynamic, we would have to use something different than the bottom sheet, I think (or use a bottom sheet that can adapt better to the content size).